volume/drivers/common/default_store_enumerator.go

```diff
@@ -184,6 +184,11 @@ func match(
 	if locator == nil {
 		return hasSubset(v.Locator.VolumeLabels, volumeLabels)
 	}
+	if locator.GetGroup() != nil {
+		if v.GetSpec().GetGroup() == nil || v.GetSpec().GetGroup() != locator.GetGroup() {
```

Comment:
I'm not sure this is correct. It is comparing the addresses of the Group objects. Instead it should be comparing the Id of the group object.

I think you should create a new file called `api/group.go` and in it add a receiver to `(* Group)` called `IsMatch` like the  one done for Ownership. Then it can do the comparison according to the values in the object.

Reply:
gotcha makes sense.